thunder/benchmarks/benchmark_peft.py

```diff
@@ -0,0 +1,724 @@
+        raise
+
+
+def setup_lora(model: torch.nn.Module, verbose: bool = False) -> torch.nn.Module:
+    """Apply LoRA to the model."""
+    if verbose:
+        logger.info("Applying LoRA to model")
+
+    lora_config = LoraConfig(
+        r=16,  # rank
+        target_modules="all-linear",  # See: https://huggingface.co/docs/peft/package_reference/lora#peft.LoraConfig.target_modules
+        lora_alpha=32,
+        task_type="CAUSAL_LM",
+    )
+
+    model = get_peft_model(model, lora_config)
+
+    # Freeze all parameters except LoRA
+    for name, param in model.named_parameters():
+        if "lora" not in name.lower():
+            param.requires_grad = False
+        else:
+            param.requires_grad = True
+            if verbose:
+                logger.info(f"Setting gradient for LoRA parameter: {name}")
+
+    if verbose:
+        logger.info("LoRA applied to model")
+
+    return model
+
+
+def setup_fsdp2(model: torch.nn.Module, devices: int, verbose: bool = False) -> torch.nn.Module:
+    """Apply FSDP2 to the model with ZeRO-3 style sharding."""
+    if devices <= 1:
+        return model
+
+    if verbose:
+        logger.info("Applying FSDP2 to model with ZeRO-3 style sharding")
+
+    mesh = DeviceMesh("cuda", torch.arange(devices), mesh_dim_names=("dp",))
+
+    # Configure model for static shapes before FSDP2
+    if hasattr(model, "config"):
+        model.config.use_cache = True
+        model.config.max_position_embeddings = args.seq_length
+        if verbose:
+            logger.info(f"Configured model for static shapes with sequence length: {args.seq_length}")
+
+    # Apply FSDP2 with ZeRO-3 style sharding
+    # First wrap individual layers
+    for name, module in model.named_modules():
+        if isinstance(module, (torch.nn.Linear, torch.nn.Embedding)):
+            if verbose:
+                logger.info(f"Wrapping layer {name} with FSDP2")
+            fully_shard(module, mesh=mesh)
+
+    # Then wrap the entire model
+    model = fully_shard(
+        model,
+        mesh=mesh,
+        reshard_after_forward=True,
+    )
+    if verbose:
+        logger.info("FSDP2 applied to model")
+
+    return model
+
+
+def setup_compilation(model: torch.nn.Module, backend: str, verbose: bool = False) -> torch.nn.Module:
+    """Apply compilation settings to the model."""
+    if backend == "torchjit":
+        logger.info("Compiling model with torch.compile")
+        dist_print("Resetting cache size for torch.compile")
+        import torch._dynamo.config as dynamo_config
+
+        # Fixes recompilation issues with inductor
+        dynamo_config.cache_size_limit = 64
+        model = torch.compile(model)
+    elif backend == "thunder":
+        import thunder
+        import thunder.dynamo
+        import torch._dynamo.config as dynamo_config
+        from thunder.dev_utils.nvtx_profile_transform import NvtxProfileTransform
+        from thunder.executors.transformer_engineex import transformer_engine_ex
```

Comment:
I possibly miss something innegligible, but could you have all the imports at the beginning of the file?